### PR TITLE
fix: nx16 issues

### PR DESCRIPTION
**Pull request by kockicica**
remove deprecated @nx/workspace calls. should fix issues with NX version 16 and 17

### CODE REVIEW

**Comment by brnbs** [comment]
@bennymeg please consider accepting this PR. It ensures compatibility of the plugin with Nx 17, which was released almost a month ago. This update would be greatly beneficial for those using the latest version. Thank you for your time and effort.

File: packages/nx-electron/package.json [hunk 1]
```diff
@@ -47,7 +47,7 @@
   ],
   "peerDependencies": {
     "electron": "*",
-    "@nx/workspace": "^16.0.0"
+    "@nx/workspace": ">=16"
```
Comment (bennymeg) [inline packages/nx-electron/package.json:50]: Please update to "^16"
I will publish a new version for v17

File: packages/nx-electron/src/executors/build/executor.ts [hunk 2]
```diff
@@ -1,24 +1,23 @@
...
-import { BuildBuilderOptions } from '../../utils/types';
-import { getSourceRoot } from '../../utils/workspace';
-import { MAIN_OUTPUT_FILENAME } from '../../utils/config';
-import { createPackageJson } from '@nx/js';
+import {getElectronWebpackConfig} from '../../utils/electron.config';
+import {normalizeBuildOptions} from '../../utils/normalize';
+import {BuildBuilderOptions} from '../../utils/types';
+import {getSourceRoot} from '../../utils/workspace';
+import {MAIN_OUTPUT_FILENAME} from '../../utils/config';
+import {createPackageJson} from '@nx/js';
```
Comment (bennymeg) [inline packages/nx-electron/src/executors/build/executor.ts:20]: Please restore the spacing between the { brackets }

File: packages/nx-electron/src/executors/execute/executor.ts [hunk 3]
```diff
@@ -161,15 +161,17 @@ async function* startBuild(
...
     {
       ...options.buildTargetOptions,
       generatePackageJson: false,
       watch: options.watch,
     },
     context
-  );
+  )) {
+    yield res;
+  }
```
Comment (bennymeg) [inline packages/nx-electron/src/executors/execute/executor.ts:174]: Please rename `res` to `buildEventResponse`

File: packages/nx-electron/src/executors/build/executor.ts [hunk 4]
```diff
@@ -126,6 +126,12 @@ export function executor(
       })
     )
   );
+  for await (const res of results) {
+    if(!res.success) {
+      return res;
+    }
+  }
+  return {success: true};
```
Comment (bennymeg) [inline packages/nx-electron/src/executors/build/executor.ts]: Please rename `res` to `result`

**Review by bennymeg — COMMENTED** [review]
Thanks for the contribution. 
Could you please address the following review?

File: packages/nx-electron/src/executors/build/executor.ts (hunk 4 above)
Comment (kockicica) [reply packages/nx-electron/src/executors/build/executor.ts]: i think this part does not exist anymore

**Review by bennymeg — APPROVED** [review]
Thanks!

File: packages/nx-electron/package.json (hunk 1 above)
Comment (brnbs) [reply packages/nx-electron/package.json]: @bennymeg Is there anything I can assist with to publish the new version? Due to this change, I am unable to use the package with Nx 17

Comment (kockicica) [reply packages/nx-electron/package.json]: there is @kockicica/nx-electron npm package as quick and dirty fix until proper nx17 version is released.

Comment (bennymeg) [reply packages/nx-electron/package.json]: @brnbs could you try `nx-electron@17.0.0-alpha.0`?

Comment (brnbs) [reply packages/nx-electron/package.json]: @bennymeg 17.0.0-alpha.0 seems to work for me with Nx 17.0.2
@kockicica thanks for sharing!

**Pull request merged**